lib/src/wallet/keys/unified.rs

```diff
@@ -0,0 +1,406 @@
+use std::{
+    collections::{HashMap, HashSet},
+    io::{self, Read, Write},
+};
+
+use bip0039::Mnemonic;
+use byteorder::{ReadBytesExt, WriteBytesExt};
+use orchard::keys::Scope;
+
+use zcash_client_backend::address::UnifiedAddress;
+use zcash_encoding::Vector;
+use zcash_primitives::{legacy::TransparentAddress, zip32::DiversifierIndex};
+use zingoconfig::ZingoConfig;
+
+use crate::wallet::traits::ReadableWriteable;
+
+use super::{extended_transparent::KeyIndex, get_zaddr_from_bip39seed, ToBase58Check};
+
+#[derive(Clone, Debug)]
+pub struct UnifiedSpendAuthority {
+    orchard_key: orchard::keys::SpendingKey,
+    sapling_key: zcash_primitives::zip32::ExtendedSpendingKey,
```

Comment:
Why aren't these fields `Option<..>`?

Reply:
Because by default we generate a single UA with all three pools... So we need all three keys. We don't currently have a use-case for _not_ including all key types, and the logic is simpler when we can rely on their existence.